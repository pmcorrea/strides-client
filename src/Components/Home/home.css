
ul {
	/* border: dashed black 2px; */
	width: 100%;
	margin: 0 auto;
	margin-bottom: 80px;
	padding: 0px;
}

li {
	width: 100%;
	padding: 10px;
	margin: 20px auto;

	border-radius: 10px;
	/* border: solid black 2px; */
	box-shadow: 0px 5px 10px rgb(0, 0, 0);
	background-image: url("habit-background.jpg");
	background-size: cover;
	list-style: none;
}
.habit-tool-bar>div>button>a {
	color: black;
}

li>p {
	font-size: 1.2rem;
	font-weight: 300;
	color: white;
}

.habit-tool-bar {
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: center;
}

.habit-tool-bar>p {
	margin: 0;
	font-size: 1.2rem;
	font-weight: 600;
	color: white;
}

.habit-tool-bar>div {
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: center;
	margin: 0;
}

.habit-tool-bar>div>button {
	margin: 0 5px;
	width: 30px;
	height: 30px;
	border: none;
	border-radius: 50%;
	background-color: white;
	font-weight: 600;

	text-align: center;
	line-height: 1rem;
}

.progress-bar {
	margin: 5px 0px;
	background-color: rgb(255, 255, 255);
	border-radius: 5px;
	height: 10px;
}

.progress-bar-fill {
	background-color: rgb(45, 209, 45);
	/* width: 30%; */
	height: 10px;
	margin: 0px;
	border-radius: 5px;
}

.bottom-toolbar {
	width: 80vw;
	max-width: 400px;
	height: 50px;
	padding: 10px;
	background-color: white;

	display: flex;
	flex-direction: row;
	justify-content: space-evenly;
	align-items: center;

	position: fixed;
	bottom: 0px;

	border-radius: 15px 15px 0px 0px;
	box-shadow: 0px 5px 10px rgb(0, 0, 0);;
}

.bottom-toolbar>button {
	height: 50px;
	margin: 0 auto;
	background-color: transparent;
	font-size: var(--button-font-size);
	padding: 5px;
	border: none;
}

.icon-bar-icon {
	height: 25px;
	width: 25px;
	background-color: transparent;
	border: none;
}