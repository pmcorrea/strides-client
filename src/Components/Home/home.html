<!DOCTYPE html>
<html lang="en">

	<head>
		<meta charset="UTF-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<link rel="stylesheet" href="../Login/index.css" type="text/css">
		<link rel="stylesheet" href="home.css" type="text/css">
		<title>Strides</title>
	</head>

	<body>
		<ul>
			<li>
				<div className="habit-tool-bar">
					<p>To Do</p>

					<div>
						<button>log</button>
						<button>...</button>
					</div>
				</div>

				<p>everyday</p>
				<div className="progress-bar">
					<div className="progress-bar-fill"></div>
				</div>
			</li>

			<li>
				<div className="habit-tool-bar">
					<p>To Do</p>
			
					<div>
						<button>log</button>
						<button>...</button>
					</div>
				</div>
			
				<p>everyday</p>
				<div className="progress-bar">
					<div className="progress-bar-fill"></div>
				</div>
			</li>

			<li>
				<div className="habit-tool-bar">
					<p>To Do</p>
			
					<div>
						<button>log</button>
						<button>...</button>
					</div>
				</div>
			
				<p>everyday</p>
				<div className="progress-bar">
					<div className="progress-bar-fill"></div>
				</div>
			</li>

			<li>
				<div className="habit-tool-bar">
					<p>To Do</p>
			
					<div>
						<button>log</button>
						<button>...</button>
					</div>
				</div>
			
				<p>everyday</p>
				<div className="progress-bar">
					<div className="progress-bar-fill"></div>
				</div>
			</li>


			<li>
				<div className="habit-tool-bar">
					<p>To Do</p>
			
					<div>
						<button>log</button>
						<button>...</button>
					</div>
				</div>
			
				<p>everyday</p>
				<div className="progress-bar">
					<div className="progress-bar-fill"></div>
				</div>
			</li>

			<li>
				<div className="habit-tool-bar">
					<p>To Do</p>
			
					<div>
						<button>log</button>
						<button>...</button>
					</div>
				</div>
			
				<p>everyday</p>
				<div className="progress-bar">
					<div className="progress-bar-fill"></div>
				</div>
			</li>

			<li>
				<div className="habit-tool-bar">
					<p>To Do</p>
			
					<div>
						<button>log</button>
						<button>...</button>
					</div>
				</div>
			
				<p>everyday</p>
				<div className="progress-bar">
					<div className="progress-bar-fill"></div>
				</div>
			</li>

	
		</ul>

		<div className="bottom-toolbar">
			<button><img src="../Assets/home-icon.svg" alt="" className="icon-bar-icon"></button>
			<button><img src="../Assets/plus-icon.svg" alt="" className="icon-bar-icon"></button>
			<button><img src="../Assets/profile-icon.svg" alt="" className="icon-bar-icon"></button>
		</div>
	</body>

</html>