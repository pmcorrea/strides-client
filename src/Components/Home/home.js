import React, { useState } from 'react';
import { Link } from "react-router-dom";

import HomeIcon from "../../Assets/home-icon.svg"
import PlusIcon from "../../Assets/plus-icon.svg"
import ProfileIcon from "../../Assets/profile-icon.svg"

import { useQuery, useMutation, useLazyQuery } from '@apollo/client'
import { habitById, logHabit, deleteHabit, habitsByUser } from '../../Queries/queries'

import "./home.css"
const dateHelper = require('date-fns')

function returnDays({ sunday, monday, tuesday, wednesday, thursday, friday, saturday }) {
	let days = []

	if (sunday === 'true') {
		days.push('sun')
	}

	if (monday === 'true') {
		days.push('mon')
	}

	if (tuesday === 'true') {
		days.push('tue')
	}

	if (wednesday === 'true') {
		days.push('wed')
	}

	if (thursday === 'true') {
		days.push('thu')
	}

	if (friday === 'true') {
		days.push('fri')
	}

	if (saturday === 'true') {
		days.push('sat')
	}

	if (days.length === 7) {
		return "everyday"
	}

	if (days.length === 0) {
		return "any day"
	}

	days = days.join(", ")
	return days
}



	


export default function Home() {
	const [stateError, setError] = useState(null)
	const { loading, data } = useQuery(habitsByUser)

	const [createLog, { loadinglogDay, datalogday, error }] = useMutation(
		logHabit,
		{
			onError: (error) => {
				console.log(error)
				setError(`${error}`)
			},
			refetchQueries: [{
				query: habitsByUser,
			}]
		});

	function diff(startDate) {
		console.log(startDate)
		let today_iso = new Date().toISOString()
		let today = dateHelper.parseISO(today_iso)
		startDate = dateHelper.parseISO(startDate)

		let diff = dateHelper.differenceInCalendarDays(
			today,
			startDate
		)

		diff = diff.toString()
		return diff
	}

	function progressBarPercentage(someDate) {

		let today_iso = new Date().toISOString()
		let today = dateHelper.parseISO(today_iso)
		let startDate = dateHelper.parseISO(someDate)

		let diffProgress = dateHelper.differenceInCalendarDays(
			today,
			startDate
		)


		if (diffProgress >= 30) {
			return 100
		} else if (diffProgress < 30) {
			return ((diffProgress) / 30) * 100
		}
	}
	

	return loading ? (" ") : (
		<div>
			<ul>
				{(data === undefined) ? (" ") : 

					data['habitsByUser'].map((item) => (
					<li key={item.id}>
						<div className="habit-tool-bar">
							<p>{item.title}</p>

							<div>
									<button className="cursor" onClick={() => createLog({
										variables: {
										id: `${item.id}`,
										column: diff(item.habit_start_date),
										current_streak: item.current_streak,
										last_log: item.last_log,
										highest_streak: item.highest_streak,

										habit_start_date: item.habit_start_date,
										perfect_streak: item.perfect_streak,
										sunday: item.sunday,
										monday: item.monday,
										tuesday: item.tuesday,
										wednesday: item.wednesday,
										thursday: item.thursday,
										friday: item.friday,
										saturday: item.saturday,
										last_scheduled_logged: item.last_scheduled_logged
										}})}>log</button>
								<button>
									<Link to={`/habit/${item.id}`}>
										...
									</Link>
								</button>
							</div>
						</div>

							<p>{returnDays(item)}</p>
						<div className="progress-bar">
								<div className="progress-bar-fill" style={{
									width: `${progressBarPercentage(item.habit_start_date)}%`
								}}></div>
						</div>
					</li>
				)
				)}
			</ul>

			<div className="bottom-toolbar">
				<button>
					<Link to="/home">
						<img src={HomeIcon} alt="" className="icon-bar-icon" />
					</Link>
				</button>

				<button>
					<Link to="/add-habit">
						<img src={PlusIcon} alt="" className="icon-bar-icon" />
					</Link>
				</button>

				<button>
					<Link to="/profile">
						<img src={ProfileIcon} alt="" className="icon-bar-icon" />
					</Link>
				</button>
			</div>	
		</div>
	);
}