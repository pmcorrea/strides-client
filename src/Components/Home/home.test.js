import React from 'react';
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import Home from './home';

afterEach(cleanup)

// Snapshot Test
it ('should take a snapshot', () => {
	const { asFragment } = render(
	<BrowserRouter>
			<Home />
	</BrowserRouter>
	)

	expect(asFragment(
		<BrowserRouter>
			<Home />
		</BrowserRouter>)
		).toMatchSnapshot()
})

// Smoke Test
it('renders without crashing', () => {
	const div = document.createElement('div')

	ReactDOM.render(
		<BrowserRouter>
			<Home />
		</BrowserRouter>,
		div
	)
	
	ReactDOM.unmountComponentAtNode(div)
})