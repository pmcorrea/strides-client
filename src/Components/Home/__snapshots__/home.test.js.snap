// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <ul>
      <li>
        <div
          class="habit-tool-bar"
        >
          <p>
            To Do
          </p>
          <div>
            <button>
              log
            </button>
            <button>
              <a
                href="/habit"
              >
                ...
              </a>
            </button>
          </div>
        </div>
        <p>
          everyday
        </p>
        <div
          class="progress-bar"
        >
          <div
            class="progress-bar-fill"
          />
        </div>
      </li>
      <li>
        <div
          class="habit-tool-bar"
        >
          <p>
            To Do
          </p>
          <div>
            <button>
              log
            </button>
            <button>
              <a
                href="/habit"
              >
                ...
              </a>
            </button>
          </div>
        </div>
        <p>
          everyday
        </p>
        <div
          class="progress-bar"
        >
          <div
            class="progress-bar-fill"
          />
        </div>
      </li>
      <li>
        <div
          class="habit-tool-bar"
        >
          <p>
            To Do
          </p>
          <div>
            <button>
              log
            </button>
            <button>
              <a
                href="/habit"
              >
                ...
              </a>
            </button>
          </div>
        </div>
        <p>
          everyday
        </p>
        <div
          class="progress-bar"
        >
          <div
            class="progress-bar-fill"
          />
        </div>
      </li>
      <li>
        <div
          class="habit-tool-bar"
        >
          <p>
            To Do
          </p>
          <div>
            <button>
              log
            </button>
            <button>
              <a
                href="/habit"
              >
                ...
              </a>
            </button>
          </div>
        </div>
        <p>
          everyday
        </p>
        <div
          class="progress-bar"
        >
          <div
            class="progress-bar-fill"
          />
        </div>
      </li>
      <li>
        <div
          class="habit-tool-bar"
        >
          <p>
            To Do
          </p>
          <div>
            <button>
              log
            </button>
            <button>
              <a
                href="/habit"
              >
                ...
              </a>
            </button>
          </div>
        </div>
        <p>
          everyday
        </p>
        <div
          class="progress-bar"
        >
          <div
            class="progress-bar-fill"
          />
        </div>
      </li>
      <li>
        <div
          class="habit-tool-bar"
        >
          <p>
            To Do
          </p>
          <div>
            <button>
              log
            </button>
            <button>
              <a
                href="/habit"
              >
                ...
              </a>
            </button>
          </div>
        </div>
        <p>
          everyday
        </p>
        <div
          class="progress-bar"
        >
          <div
            class="progress-bar-fill"
          />
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
